docs/build/workflow-snapshots.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Workflow Snapshots
+sidebar_label: Workflow Snapshots
+slug: /workflow-snapshots
+---
+
+Workflow snapshots capture and store the state of a workflow and job code at a specific point in time based on a save operation and/or run. Snapshots are helpful with debugging, auditing, and improving overall workflow performance.
+
+### When is a snapshot made?
+Snapshots are created in 2 ways:
+1. When a user saves changes to their workflow either through the canvas or the job editor
+2. When a run is made either by creating a new work order or retrying a run
+
+### How can I view a snapshot?
+To view a snapshot, go to the `History` page. Expand a work order to see the runs included. 
+
+![Snapshot1](/img/snapshots1.png)
+
+From the expanded history view, there are two ways to visualize snapshots: 
+1. By inspecting a step in the run
+2. From the Run view
+
+#### Viewing a snapshot by inspecting a step in the run
+
+Click on the inspect icon in front of the step you would like to view. 
+
+![Inspect](/img/inspect.png)
+
+This will open the inspector screen for that step in the run with all associated artifacts: logs and input/output data. On the inspector, you'll notice that you're in a read-only mode, and hovering on the workflow snapshot ID chip, you’ll see a message that reads “You are viewing a snapshot of this workflow that was taken on ….”
```

Comment:
Should we link to this existing docs page for more on the Inspector? 
https://docs.openfn.org/documentation/build/steps/step-editor